src/main/java/com/wanderersoftherift/wotr/gui/menu/RuneAnvilMenu.java

```diff
@@ -148,9 +148,11 @@ public void apply() {
             GearSocket currentSocket = currentSockets.sockets().get(i);
             ItemStack runegem = slot.getItem();
             RunegemData runegemData = runegem.get(WotrDataComponentType.RUNEGEM_DATA);
+            List<GearSocket> existingSockets = new ArrayList<>(newSockets);
```

Comment:
Are there situations where a runegem cannot apply at all because all its possibilities have already been applied? I guess that might be a separate issue.

Reply:
Right now, these situations indeed exist. And they are rather common. Health runegem, for example, can not apply to weapons. Once we have more stuff, like leech and armor penetration, the worst ones in this will be better.